From 87bb73db5fabc75ea34c214c157a74fa4ab7c265 Mon Sep 17 00:00:00 2001
From: fundawang <fundawang@yeah.net>
Date: Thu, 27 Aug 2026 09:29:17 +0800
Subject: [PATCH] only use AV_PIX_FMT_D3D12 in ffmpeg 7

As we ensure freerdp builds on ffmpeg 5.1.3, make it conditioned.
---
 libfreerdp/codec/h264_ffmpeg.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/libfreerdp/codec/h264_ffmpeg.c b/libfreerdp/codec/h264_ffmpeg.c
index 8ebf2a08..d04162a3 100644
--- a/libfreerdp/codec/h264_ffmpeg.c
+++ b/libfreerdp/codec/h264_ffmpeg.c
@@ -301,7 +301,9 @@ static const char* av_format_str(int32_t format)
 		EVCASE(AV_PIX_FMT_P412LE);
 		EVCASE(AV_PIX_FMT_GBRAP14BE);
 		EVCASE(AV_PIX_FMT_GBRAP14LE);
+#if LIBAVUTIL_VERSION_INT >= AV_VERSION_INT(59,16,100)
 		EVCASE(AV_PIX_FMT_D3D12);
+#endif
 		EVCASE(AV_PIX_FMT_NB);
 		default:
 			return "AV_PIX_FMT_UNKNOWN";
-- 
2.50.1 (Apple Git-155)

